### DISCUSSION TITLE
Gen8MemoryMissingHT: Fishy Vs. Invaild

### DISCUSSION THREAD

**Discussion by DarkKrimson** [kwsch/PKHeX#4290] [Q&A]
Pokemon Details:
- Legit and completely unedited
- Caught in 2019 in SWSH
- Was transferred to HOME and was transferred back to SWSH. (Not sure when it was originally transferred over but could have been around early HOME release)

PKHeX Details:
- When viewed in PKHeX with `Gen8MemoryMissingHT=Fishy` it passes.
- When viewed in PKHeX with `Gen8MemoryMissingHT=Invalid` it fails for `Invalid: Memory: Handling Trainer Memory missing.`
- PKHeX detects the pokemon currently with it's OT. not HT.
- Both Pokémon HOME's OT and SWSH share the same OT name.
- Memories with OT is `When Dracozolt heard a rumor that ASH arrived at a place... in a vast field, it imagined how ASH looked there. The Pokémon definitely remembers that it got emotional.`
- Memories with notOT: `Dracozolt seems to have a fond memory, but it doesn’t seem to be able to remember the details...`

-----
The only ways I could get it to pass legality are 1 of 2 methods:
1. Remove HT completely like it was never traded. _Although it would still have a HOME tracker which makes this way illegal despite PKHeX saying it's legal._
2. Copying `Memories with OT` details to `Memories with notOT`. _Not sure if this correct either as this is an edit and HOME wouldn't understand why these values changed since it was last seen._

I have other Pokémon that were transferred to HOME at the same time with this one and have the same memory details and pass legality. I'm having trouble finding a reason why some flag as illegal and some flag as legal even though they both share very similar details (has OT memory details and none for notOT details)

It seems like HOME didn't correctly update/address the memories when sending back to it's origin game SWSH.

Is this something that HOME is doing when updating Pokémon on the return trip back or do I just need to make the edits explained in method 2?

**Comment by Lusamine** [comment]
Without seeing the Pokémon, it's hard to tell what the situation is that is causing it to flag. In general, if you have not externally edited a Pokémon caught on an unmodded game, it is legitimate regardless of what any program or person says.

The expected behavior should not be that you have to edit legitimate Pokémon to fit PKHeX's legality checks. If a legitimate Pokémon is being flagged, it should be reported as a bug that can be fixed under [Issues](https://github.com/kwsch/PKHeX/issues).

That said, it is possible that if you viewed the Pokémon in PKHeX when there was any sort of error, it could have been edited to be incorrect. For example: https://github.com/kwsch/PKHeX/issues/4227 if you had SlotWrite > SetUpdatePKM on, it would have updated the HT memory to be incorrect.

**Reply by DarkKrimson** [reply]
Checking the original backup save file, it appears some mons were being set to HT as CurrentHandler but should have not been.
I transferred all effected mons to HOME to see if this would resolve the issue on return. It did for most of them but not for a few like I mentioned in the initial post.

HOME seemed to just change the CurrentHandler to OT upon arrival back to SWSH. This corrected a few in PKHeX but there are still a few remaining that are still are flagged where changing CurrentHandler to OT was not enough. These are the same ones in question from the initial post.

This appears to line up with your last statement.

---

Comparing the backup, pre-HOME transfer fix save, post-HOME transfer fix save,

I have come to some conclusions:
- Only Pokemon that have a HOME Tracker were the Pokemon that had HT as the CurrentHandler.
- Any of the Pokemon with HT as CurrentHandler were gaining friendship in HT and post HOME transfer fix, they were correctly set back to OT and it's friendship values but only a select few are still flagged.

Its strange how the backup has CurrentHandler issues that appear to originate from a HOME transfer way back. I cannot find anything else that would cause these Pokemon to still remain flagged. Is there extra bytes or other data that could cause it to be flagged?

Extra info: The mons in question are all four fossils, Blacephalon from DA, Ingame gift Silvally. All acquired legitimately, and unedited. I don't think there is a correlation between but throwing it out there just in case.

**Reply by Lusamine** [reply]
Sorry, as I said before, without the files, it's just a guessing game at what's flagging and why. If you believe they are truly legitimate, unmodified, and erroneously flagged, you should open an issue so this can be fixed. I won't be guessing blindly since it's not a good use of my time and doesn't improve the program.

If you're trying to fix them because they might have been accidentally edited, my suggestion would be to revert to a backup.

If you can provide ANY data at all for mons that are flagging vs mons that are not, that would help provide a more specific answer.
